packages/legend-application-query/src/components/QueryEditor.tsx

```diff
@@ -84,103 +84,107 @@ import { QUERY_DOCUMENTATION_KEY } from '../application/LegendQueryDocumentation
 import { debounce } from '@finos/legend-shared';
 import { LegendQueryTelemetryHelper } from '../__lib__/LegendQueryTelemetryHelper.js';
 import { QUERY_EDITOR_TEST_ID } from '../__lib__/LegendQueryTesting.js';
+import type { Query } from '@finos/legend-graph';
 
-const CreateQueryDialog = observer(() => {
-  const editorStore = useQueryEditorStore();
-  const createQueryState = editorStore.queryCreatorState;
-  const close = (): void => {
-    createQueryState.setShowCreateModal(false);
-    createQueryState.editorStore.setExistingQueryName(undefined);
-  };
-  const applicationStore = useApplicationStore();
-  const create = (): void => {
-    flowResult(createQueryState.createQuery()).catch(
-      applicationStore.alertUnhandledError,
+const CreateQueryDialog = observer(
```

Comment:
See my comment in `QueryCreatorState`, I think we pass in the query eagerly than passing in the component here